Introduce public MergedDataMap

lucylq opened the pull request:
Summary:
Add public merged data map. Module can use this to resolve multiple named data maps.
Creating as a sep dependency rather than inside module/ so it can be used independently of module. (think there may be some other internal usages soon)

Add support for BUCK and CMake. On CMake side, have to add the dep for everyone who uses module ..

Reviewed By: swolchok

Differential Revision: D83527299

swolchok reviewed (approved):
Review automatically exported from Phabricator review in Meta.

lucylq committed:
Introduce public MergedDataMap (#14861)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/14861

Add public merged data map. Module can use this to resolve multiple named data maps.
Creating as a sep dependency rather than inside module/ so it can be used independently of module. (think there may be some other internal usages soon)

Add support for BUCK and CMake. On CMake side, have to add the dep for everyone who uses module ..

Reviewed By: swolchok

Differential Revision: D83527299
